### PR TITLE
Do not try to render bad client metrics

### PR DESCRIPTION
**Pull request by esbie**
Related to #1287

Gracefully handle the js edge case where client metric names are not
formatted as we expect.

### CODE REVIEW

**Comment by adleong** [comment]
I notice that "dispatcher" is in `metrics.js`.  Is it possible to add a test that catches this error?

**Comment by esbie** [comment]
Yes I noticed that too! I'm a little embarrassed to admit we don't have tests for this component yet. It is certainly possible to do, would you be alright adding tests in a followup PR?

**Comment by adleong** [comment]
Mostly I just was looking for a way of testing that the issue is resolved.  If you've tested manually and are confident then 🚀.

File: admin/src/main/resources/io/buoyant/admin/js/src/router_clients.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ define([
         return initializeClient(client);
       });
 
+      routerClients = _.compact(routerClients);
```
Comment (rmars) [inline admin/src/main/resources/io/buoyant/admin/js/src/router_clients.js]: tioli: `_.chain()` this to the map in the previous line so these don't get separated
